testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeAdls.java

```diff
@@ -0,0 +1,103 @@
+import static java.nio.file.attribute.PosixFilePermissions.fromString;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvMultinodeAdls
+        extends EnvironmentProvider
+{
+    private final DockerFiles.ResourceProvider configDir;
+    private final String hadoopImagesVersion;
+
+    @Inject
+    public EnvMultinodeAdls(DockerFiles dockerFiles, StandardMultinode standardMultinode, Hadoop hadoop, EnvironmentConfig environmentConfig)
+    {
+        super(ImmutableList.of(standardMultinode, hadoop));
+        configDir = requireNonNull(dockerFiles, "dockerFiles is null").getDockerFilesHostDirectory("conf/environment/multinode-adls");
+        hadoopImagesVersion = requireNonNull(environmentConfig, "environmentConfig is null").getHadoopImagesVersion();
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        String dockerImageName = "ghcr.io/trinodb/testing/hdp3.1-hive:" + hadoopImagesVersion;
+
+        builder.configureContainer(HADOOP, container -> {
+            container.setDockerImageName(dockerImageName);
+            container
+                    .withCopyFileToContainer(forHostPath(getCoreSiteXml()), "/etc/hadoop/conf/core-site.xml");
+        });
+
+        builder.configureContainer(COORDINATOR, container -> container
+                .withEnv("ABFS_ACCOUNT", requireEnv("ABFS_ACCOUNT"))
+                .withEnv("ABFS_ACCESS_KEY", requireEnv("ABFS_ACCESS_KEY")));
+
+        builder.configureContainer(WORKER, container -> container
+                .withEnv("ABFS_ACCOUNT", requireEnv("ABFS_ACCOUNT"))
+                .withEnv("ABFS_ACCESS_KEY", requireEnv("ABFS_ACCESS_KEY")));
+
+        builder.configureContainer(TESTS, container -> container
+                .withEnv("ABFS_CONTAINER", requireEnv("ABFS_CONTAINER"))
+                .withEnv("ABFS_ACCOUNT", requireEnv("ABFS_ACCOUNT")));
+
+        builder.addConnector("hive", forHostPath(configDir.getPath("hive.properties")));
+    }
+
+    private Path getCoreSiteXml()
+    {
+        try {
+            String coreSite = Files.readString(configDir.getPath("core-site-abfs-template.xml"))
+                    .replace("%ABFS_ACCOUNT%", requireEnv("ABFS_ACCOUNT"))
+                    .replace("%ABFS_ACCESS_KEY%", requireEnv("ABFS_ACCESS_KEY"));
+            File coreSiteXml = Files.createTempFile("core-site", ".xml", PosixFilePermissions.asFileAttribute(fromString("rwxrwxrwx"))).toFile();
+            coreSiteXml.deleteOnExit();
+            Files.writeString(coreSiteXml.toPath(), coreSite);
+            return coreSiteXml.toPath();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static String requireEnv(String variable)
+    {
+        return requireNonNull(System.getenv(variable), "environment variable not set: " + variable);
```

Comment:
format the message only when confition not satisifed

perhaps sth like

```suggestion
        return requireNonNull(System.getenv(variable), () -> "environment variable not set: " + variable);
```